ext/webserial/01_webserial.js

```diff
@@ -0,0 +1,397 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/// <reference path="../../core/internal.d.ts" />
+
+((window) => {
+  const core = window.Deno.core;
+  const webidl = window.__bootstrap.webidl;
+  const { DOMException } = window.__bootstrap.domException;
+  const {
+    Promise,
+    PromiseResolve,
+    PromiseAll,
+    Symbol,
+    TypeError,
+    Uint8Array,
+  } = window.__bootstrap.primordials;
+
+  webidl.converters.SerialPortRequestOptions = webidl.createDictionaryConverter(
+    "SerialPortRequestOptions",
+    [
+      {
+        key: "filters",
+        converter: webidl.createSequenceConverter(
+          webidl.converters.SerialPortFilter,
+        ),
+      },
+    ],
+  );
+
+  webidl.converters.SerialPortFilter = webidl.createDictionaryConverter(
+    "SerialPortFilter",
+    [
+      {
+        key: "usbVendorId",
+        converter: webidl.converters["unsigned short"],
+      },
+      {
+        key: "usbProductId",
+        converter: webidl.converters["unsigned short"],
+      },
+    ],
+  );
+
+  webidl.converters.ParityType = webidl.createEnumConverter("ParityType", [
+    "none",
+    "even",
+    "odd",
+  ]);
+
+  webidl.converters.FlowControlType = webidl.createEnumConverter(
+    "FlowControlType",
+    [
+      "none",
+      "hardware",
+    ],
+  );
+
+  webidl.converters.SerialOptions = webidl.createDictionaryConverter(
+    "SerialOptions",
+    [
+      {
+        key: "baudRate",
+        converter: (V, opts) =>
+          webidl.converters["unsigned long"](V, {
+            ...opts,
```

Comment:
Not sure if the safe iterators PR also applies to object spreads. If so, then these'll need primordials as well.

Reply:
yes, this PR is extremely outdated, soa lot of things here are quite wrong